libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -333,26 +333,19 @@ void CompilerUtils::encodeToMemory(
 )
 {
 	// stack: <v1> <v2> ... <vn> <mem>
+	bool const encoderV2 = m_context.experimentalFeatureActive(ExperimentalFeature::ABIEncoderV2);
 	TypePointers targetTypes = _targetTypes.empty() ? _givenTypes : _targetTypes;
 	solAssert(targetTypes.size() == _givenTypes.size(), "");
 	for (TypePointer& t: targetTypes)
 	{
-		solUnimplementedAssert(
-			t->mobileType() &&
-			t->mobileType()->interfaceType(_encodeAsLibraryTypes) &&
-			t->mobileType()->interfaceType(_encodeAsLibraryTypes)->encodingType(),
-			"Encoding type \"" + t->toString() + "\" not yet implemented."
-		);
-		t = t->mobileType()->interfaceType(_encodeAsLibraryTypes)->encodingType();
+		TypePointer tEncoding = t->fullEncodingType(_encodeAsLibraryTypes, encoderV2, !_padToWordBoundaries);
```

Comment:
What purpose does ``tEncoding`` serve? Why not directly move into ``t`` and assert ``t`` instead below?

Reply:
If we overwrite `t`, we cannot use it in the assertion message...

Reply:
Ah, I hadn't seen that it's used there.